commons/src/main/java/com/powsybl/commons/datasource/DataSourceUtil.java

```diff
@@ -18,7 +18,8 @@ public interface DataSourceUtil {
 
     static String getFileName(String baseName, String suffix, String ext) {
         Objects.requireNonNull(baseName);
-        return baseName + (suffix != null ? suffix : "") + (ext == null || ext.isEmpty() ? "" : "." + ext);
+        String end = (suffix != null ? suffix : "") + (ext == null || ext.isEmpty() ? "" : "." + ext);
```

Comment:
Taking the suffix into account in `baseName.endsWith(end)` is confusing things... Better only test `baseName.endsWith("." + ext)`.

And in fact we could do a bit better in case of `baseName.endsWith("." + ext)` with a non-null suffix. But I suppose this is useless?